src/core_plugins/kibana/public/visualize/listing/visualize_listing.js

```diff
@@ -0,0 +1,167 @@
+      calculateItemsOnPage();
+    });
+  };
+
+  this.items = [];
+  this.pageOfItems = [];
+  this.filter = '';
+
+  this.pager = pagerService.create(this.items.length, 20, 1);
+
+  /**
+   * Remember sort direction per property.
+   */
+  this.sortProperties = [{
+    name: 'title',
+    getValue: item => item.title,
+    isSelected: true,
+    isAscending: true,
+  }, {
+    name: 'type',
+    getValue: item => item.type.title,
+    isSelected: false,
+    isAscending: true,
+  }];
+
+  this.getSortProperty = function getSortProperty() {
+    return _.find(this.sortProperties, property => property.isSelected);
+  };
+
+  this.getSortPropertyByName = function getSortPropertyByName(name) {
+    return _.find(this.sortProperties, property => property.name === name);
+  };
+
+  this.isAscending = function isAscending() {
+    const sortProperty = this.getSortProperty();
+    return sortProperty.isAscending;
+  };
+
+  this.sortOn = function sortOn(propertyName) {
+    const sortProperty = this.getSortProperty();
+
+    if (sortProperty.name === propertyName) {
+      sortProperty.isAscending = !sortProperty.isAscending;
+    } else {
+      sortProperty.isSelected = false;
+      this.getSortPropertyByName(propertyName).isSelected = true;
+    }
+
+    calculateItemsOnPage();
+  };
+
+  this.toggleAll = function toggleAll() {
+    if (this.areAllItemsChecked()) {
+      selectedItems = [];
+    } else {
+      selectedItems = this.items.slice(0);
+    }
+  };
+
+  this.toggleItem = function toggleItem(item) {
+    if (this.isItemChecked(item)) {
+      const index = selectedItems.indexOf(item);
+      selectedItems.splice(index, 1);
+    } else {
+      selectedItems.push(item);
+    }
+  };
+
+  this.isItemChecked = function isItemChecked(item) {
+    return selectedItems.indexOf(item) !== -1;
+  };
+
+  this.areAllItemsChecked = function areAllItemsChecked() {
+    return this.getSelectedItemsCount() === this.items.length;
+  };
+
+  this.getSelectedItemsCount = function getSelectedItemsCount() {
+    return selectedItems.length;
+  };
+
+  this.deleteSelectedItems = function deleteSelectedItems() {
+    const doDelete = () => {
+      const selectedIds = selectedItems.map(item => item.id);
+
+      visualizationService.delete(selectedIds)
+        .then(fetchObjects)
+        .then(() => {
+          selectedItems = [];
+        })
+        .catch(error => notify.error(error));
+    };
+
+    confirmModal(
+      'Are you sure you want to delete the selected visualizations? This action is irreversible!',
+      {
+        confirmButtonText: 'Delete',
+        onConfirm: doDelete
+      });
+  };
+
+  this.onPageNext = () => {
+    this.pager.nextPage();
+    calculateItemsOnPage();
+  };
+
+  this.onPagePrevious = () => {
+    this.pager.previousPage();
+    calculateItemsOnPage();
+  };
+
+  this.getUrlForItem = function getUrlForItem(item) {
+    return `#/visualize/edit/${item.id}`;
+  };
+
+  $scope.$watch(() => this.filter, () => {
```

Comment:
I think this belongs with the rest of the initialization logic up top (or even better in a constructor)